src/extended-sam.yaml

```diff
@@ -0,0 +1,160 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: 'AWS::Serverless-2016-10-31'
+Parameters: 
+  SESSendFrom: 
+    Type: String
+    Description: "Email thats authorised in account for SES sending. Will not email if blank"
+  SESSendTo: 
+    Type: String
+    Description: "List of comma emails to send to"
+  SESRegion: 
+    Type: String
+    Default: "us-east-1"
+    Description: "Region you have configured SES in (Default us-east-1)"
+  S3Bucket: 
+    Type: String
+    Description: "S3 Bucket to store the report in"
+  AccountLabel: 
+    Type: String
+    Default: Email
+    Description: "Email or Name, default is Email"
+  ListOfCostTags: 
+    Type: String
+    Description: "Comma seperated list of all Cost Allocation Tags"
+  CurrentMonth:
+    Type: String
+    Description: "true | false for if current partial month included"
+  DayOfMonth:
+    Type: Number
+    Default: 6
+    Description: "DayOfMonth for Schedule, default 6 - use 12 if you want to report refunds and finalised Enterprise Support."
+  TrailingDays:
+    Type: Number
+    Default: 7
+    Description: "Number of trailing days to report on"
+
+  ProxyName:
+    Type: String
+    Default: exampleProxy
+  InstanceName:
+    Type: String
+    Default: env-prod-biodashboard-db
+  BootstrapSecretReaderRoleArn:
+    Type: String
+    Default: arn:aws:iam::074763112859:role/RDSProxyRole
+  BootstrapProxySecretArn:
+    Type: String
+    Default: arn:aws:secretsmanager:us-east-1:074763112859:secret:prod-db-main-0sQkpW
+  SubnetIds:
+    Type: String
+    Default: subnet-7732c54a,subnet-678ac84c,subnet-772c722e,subnet-f5beba82
+
+Resources:
+  CostExplorerReport:
+    Type: 'AWS::Serverless::Function'
+    Properties:
+      FunctionName: CostExplorerReportLambda
+      Description: "Function for generating / sending monthly cost report"
+      MemorySize: 256
+      Timeout: 60
+      Handler: lambda.main_handler
+      Runtime: python3.6
+      CodeUri: ../bin/lambda.zip
+      Layers:
+        - arn:aws:lambda:us-east-1:749981256976:layer:aws-cost-explorer-report:1
+      Role: !GetAtt CostExplorerReportLambdaIAMRole.Arn
+      Environment:
+        Variables:
+          TABLE_NAME: !Ref Table
+      Environment:
+        Variables:
+          S3_BUCKET: !Ref S3Bucket
+          SES_SEND: !Ref SESSendTo
+          SES_FROM: !Ref SESSendFrom
+          SES_REGION: !Ref SESRegion
+          COST_TAGS: !Ref ListOfCostTags
+          ACCOUNT_LABEL: !Ref AccountLabel
+          CURRENT_MONTH: !Ref CurrentMonth
+          INC_SUPPORT: 'false'
+          TRAILING_DAYS: !Ref TrailingDays
+      Events:
+        DailyEvent:
+          Properties:
+            Schedule: cron(0 0 * * ? *)
+          Type: Schedule
+  
+  CostExplorerReportLambdaIAMRole:
```

Comment:
I don't think you'll run into this here, but be warned: sometimes with inline IAM policies, you get confusing errors because of a race condition between IAM policy creation/syncing and when it is used for the first time. If you see errors anywhere telling you an inline IAM policy doesn't exist, suspect this root cause

Reply:
Good to know, should I create a managed policy and attach it by name?